app/src/main/java/com/example/arc_exapmle/note/NoteAdapter.kt

```diff
@@ -4,26 +4,41 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.example.arc_exapmle.R
 
-class NoteAdapter : RecyclerView.Adapter<NoteAdapter.NoteHolder>() {
+class NoteAdapter() :
+    ListAdapter<NoteUI, NoteAdapter.NoteHolder>(DIFF_CALLBACK) {
 
-    private var notes: List<NoteUI> = ArrayList()
+
+companion object {
+    val DIFF_CALLBACK: DiffUtil.ItemCallback<NoteUI> =
+        object : DiffUtil.ItemCallback<NoteUI>() {
+            override fun areItemsTheSame(oldItem: NoteUI, newItem: NoteUI): Boolean {
+                return oldItem.id == newItem.id
+            }
+
+            override fun areContentsTheSame(oldItem: NoteUI, newItem: NoteUI): Boolean {
+                return oldItem.title == newItem.title && oldItem.description == newItem.description && oldItem.priority == newItem.priority
```

Comment:
if `NoteUi` is `data class` you may just use == between two objs

Reply:
Done